controllers/flinkcluster_reconciler.go

```diff
@@ -342,6 +346,11 @@ func (reconciler *ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 	var observedJob = observed.job
 	var err error
 
+	// Update status changed via job reconciliation.
+	var newSavepointStatus *v1beta1.SavepointStatus
+	var newControlStatus *v1beta1.FlinkClusterControlStatus
+	defer reconciler.updateStatus(&newSavepointStatus, &newControlStatus)
```

Comment:
Does this mean it will always calls the API to update status even if both `newSavepointStatus` and `newControlStatus` are nil? can we add check to prevent unnecessary updates?

Reply:
https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/68c888b1c62bf3c21821cc311d40696e3e7bcdb7/controllers/flinkcluster_reconciler.go#L719-L726

Reply:
It is checked inside the function.